netty-mqtt/src/main/java/org/thingsboard/mqtt/MqttChannelHandler.java

```diff
@@ -282,17 +284,16 @@ private void handleUnsuback(MqttUnsubAckMessage message) {
     }
 
     private void handlePuback(MqttPubAckMessage message) {
-        MqttPendingPublish pendingPublish = this.client.getPendingPublishes().get(message.variableHeader().messageId());
-        if (pendingPublish == null) {
-            return;
-        }
-        pendingPublish.getFuture().setSuccess(null);
-        pendingPublish.onPubackReceived();
-        this.client.getPendingPublishes().remove(message.variableHeader().messageId());
-        pendingPublish.getPayload().release();
-        if (this.client.getCallback() != null) {
-            this.client.getCallback().onPubAck(message);
-        }
+        log.debug("{} Handling PUBACK", client.getClientConfig().getOwnerId());
```

Comment:
CONNECT/CONACK - debug level, everything else - trace.

Reply:
I fully agree that logging PUBACKs at DEBUG level would flood the logs with unnecessary information, but I left DISCONNECT at TRACE because if you want to see CONNECT events, you’d probably want to see DISCONNECT events as well.